slacktee.sh

```diff
@@ -266,343 +293,340 @@ function setup()
+	if [[ -z $attachment ]] && [[ ${#cond_color_patterns[@]} != 0 ]]; then
+		attachment="$internal_default_color"
+	fi
+}
 
 # ----------
 # Validate configurations
 # ----------
+function check_configuration() 
+{
+	if [[ $webhook_url == "" ]]; then
+		err_exit 1 "Please setup the webhook url of this incoming webhook integration."
+	fi
 
-if [[ $webhook_url == "" ]]; then
-	echo "Please setup the webhook url of this incoming webhook integration."
-	exit 1
-fi
-
-if [[ $upload_token == "" && $mode == "file" ]]; then
-	echo "Please provide the authentication token for file uploads."
-	exit 1
-fi
+	if [[ $upload_token == "" && $mode == "file" ]]; then
+		err_exit 1 "Please provide the authentication token for file uploads."
+	fi
 
-if [[ $channel == "" ]]; then
-	echo "Please specify a channel."
-	exit 1
-elif [[ ( "$channel" != "#"* ) && ( "$channel" != "@"* ) ]]; then
-	channel="#$channel"
-fi
+	if [[ $channel == "" ]]; then
+		err_exit 1 "Please specify a channel."
+	elif [[ ( "$channel" != "#"* ) && ( "$channel" != "@"* ) ]]; then
+		channel="#$channel"
+	fi
 
-if [[ -n "$icon" ]]; then
-	icon=${icon#:} # remove leading ':'
-	icon=${icon%:} # remove trailing ':'
-fi
+	if [[ -n "$icon" ]]; then
+		icon=${icon#:} # remove leading ':'
+		icon=${icon%:} # remove trailing ':'
+	fi
+}
 
 # ----------
 # Start script
 # ----------
+function main() 
+{
+	parse_args "$@"
+	setup_environment
+	check_configuration
+        trap cleanup SIGINT SIGTERM SIGKILL
+
+	text=""
+	if [[ -n "$title" || -n "$link" ]]; then
+		# Use link as title, if title is not specified
+		if [[ -z "$title" ]]; then
+			title="$link"
+		fi
 
-text=""
-if [[ -n "$title" || -n "$link" ]]; then
-	# Use link as title, if title is not specified
-	if [[ -z "$title" ]]; then
-		title="$link"
+		# Add title to filename in the file mode
+		if [[ "$mode" == "file" ]]; then
+			filetitle=$(echo "$title"|sed 's/[ /:.]//g')
+			filetitle="$filetitle-"
+		fi
+
+		if [[ -z "$attachment" ]]; then
+			if [[ "$mode" == "no-buffering" ]]; then
+				if [[ -n "$link" ]]; then
+					title="<$link|$title>: "
+				else
+					title="$title: "
+				fi
+			elif [[ "$mode" == "file" ]]; then
+				if [[ -n "$link" ]]; then
+					title="<$link|$title>"
+				fi
+			else
+				if [[ -n "$link" ]]; then
+					text="-- <$link|$title> --\n"
+				else
+					text="-- $title --\n"
+				fi
+			fi
+		fi
 	fi
 
-	# Add title to filename in the file mode
+	timestamp="$(date +'%m%d%Y-%H%M%S')"
+	filename="$tmp_dir/$filetitle$$-$timestamp.log"
+
 	if [[ "$mode" == "file" ]]; then
-		filetitle=$(echo "$title"|sed 's/[ /:.]//g')
-		filetitle="$filetitle-"
+		touch $filename
 	fi
 
-	if [[ -z "$attachment" ]]; then
-		if [[ "$mode" == "no-buffering" ]]; then
-			if [[ -n "$link" ]]; then
-				title="<$link|$title>: "
-			else
-				title="$title: "
-			fi
-		elif [[ "$mode" == "file" ]]; then
-			if [[ -n "$link" ]]; then
-				title="<$link|$title>"
+	while IFS='' read line; do
```

Comment:
Ah, good point! This may be a bug. We will look into it separately. 
I couldn't reproduce the invalid payload error, but the behavior was not the one I expected.